File: mobile/src/types/message.ts
export interface Message {
  id: string;
  content: string;
  sender_id: string;
  receiver_id: string;
  timestamp: number;
  encryption_key?: string;
  decrypted_content?: string;
  media_url?: string | undefined;
  media_type?: string | undefined;
  group_id?: string;
  message_hash?: string;
  signature?: string;
}

File: tchat/app/schemas/message.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime
from enum import Enum

class MediaType(str, Enum):
    IMAGE = "image"
    VIDEO = "video"
    AUDIO = "audio"
    DOCUMENT = "document"

class MessageBase(BaseModel):
    content: str
    receiver_id: str
    media_url: Optional[str] = None
    media_type: Optional[MediaType] = None
    group_id: Optional[str] = None

class MessageCreate(MessageBase):
    pass

class Message(MessageBase):
    id: str
    sender_id: str
    timestamp: datetime
    message_hash: Optional[str] = None
    signature: Optional[str] = None

    class Config:
        from_attributes = True
